components/ai_chat/core/browser/ollama/ollama_service.h

```diff
@@ -64,6 +66,11 @@ class OllamaService : public KeyedService, public mojom::OllamaService {
   virtual void ShowModel(const std::string& model_name,
                          ModelDetailsCallback callback);
 
+  // Set the model fetcher that will be owned by this service.
+  // This should be called after construction when the service is created
+  // through the factory.
+  void SetModelFetcher(std::unique_ptr<OllamaModelFetcher> model_fetcher);
```

Comment:
this should be set via constructor.

Reply:
https://github.com/brave/brave-core/pull/31569/commits/ed8640869d979ee2bdeaf3a65f496818b9b325fe